feat: adds csv files

jjaferson opened the pull request:
this PR adds the CSV file to install the MDC operator

jjaferson committed:
feat: adds csv files

grdryn commented on deploy/olm-catalog/mobile-developer-console-operator/0.2.6/mdc_v1alpha1_mobiledeveloperconsole_crd.yaml:
```diff
@@ -0,0 +1,28 @@
+  names:
+    plural: mobileclients
+    singular: mobileclient
+    kind: MobileClient
```
This is still the wrong CRD like before.

If you first create a file called `deploy/olm-catalog/csv-config.yaml`, with the contents:
```
crd-cr-paths:
  - deploy/crds
  - deploy/mdc_v1alpha1_mobileclient_crd.yaml
```

then it should create the correct one.

This `MobileClients` CR is an odd one, because it's not really managed by the operator (it's managed by MDC currently). Maybe for now we should copy it into an equivalent file as this one, and also maybe add it to the list of `owned` CRDs in the CSV spec? I'm not sure though, and would like to get other opinions. @philbrookes @maleck13 @wei-lee any thoughts?

Maybe adding the MobileClient CRD as an api here in the operator might get it to generate correctly? Not sure of any side effects of that, so let's not worry about that in this PR.

maleck13 replied:
So MDC itself does things with the CR and the operator doesn't do anything? Hmmm. Well I still think it makes sense to be in the operator CSV and probably owned as this way OLM will ensure it gets installed when the operator is installed.

grdryn replied:
@maleck13 yes, MDC is like an operator that runs in the browser as the user, rather than as a ServiceAccount.

grdryn commented on deploy/olm-catalog/mobile-developer-console-operator/mobile-developer-console-operator.package.yaml:
```diff
@@ -0,0 +1,4 @@
+packageName: integreatly-mobile-developer-console
+channels:
+  - name: integreatly
```
Let's have the channel name be `stable` here in this project. In the package file in the integreatly manifests, you can call it `integreatly`.

grdryn reviewed (dismissed):
As mentioned in the inline comment, the CRD is still incorrect.

Also, out of curiosity, why the new PR instead of the existing #62?

grdryn committed:
Fix CRDs in 0.2.6 CSV bundle

operator-sdk version: v0.10.1, commit: 872e7d997486bb587660fc8d6226eaab8b5c1087

- Removed existing package file to allow generation to work
- Added docs links to env vars in operator.yaml
- Generation command:
  operator-sdk olm-catalog gen-csv --csv-version=0.2.6 --update-crds \
    --csv-channel=stable --default-channel=true
- Fixed new package file (double channel bug)
- Filled out some additional optional fields in CSV
- Copied MobileClient CRD to bundle directory

grdryn commented:
@laurafitzgerald would you be able to give this a review?

@jjaferson I think if the build passes here & you are happy with the changes I've made here & @laurafitzgerald reviews, then I think this can be merged.

laurafitzgerald commented:
@grdryn if it's rebased on the changes I merged should the versions now be 0.3.0?

laurafitzgerald commented on deploy/olm-catalog/mobile-developer-console-operator/0.2.6/mobile-developer-console-operator.v0.2.6.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,269 @@
+                  value: https://docs.aerogear.org/limited-availability/downstream/sync.html
+                - name: MSS_DOCUMENTATION_URL
+                  value: https://docs.aerogear.org/limited-availability/downstream/mss.html
+                image: quay.io/aerogear/mobile-developer-console-operator:0.3.0
```
The version here is 0.3.0 The folder structure is using 0.2.6

jjaferson replied:
Currently the MDC manifest file is on 0.2.6, bumping the version would require generate another CSV file

grdryn replied:
Good catch! I was initially deliberately not going to generate the 0.3.0 one so that we could just get this PR correct and then it would be a simple generation for 0.3.0 in a different PR. However, once I rebased onto the latest master, this image kept getting put in here like this (I manually modified it back, but I must have regenerated again to get it back in).

Given that this is the only new comment on this PR, I'll both fix this and generate the 0.3.0 in this PR now.

grdryn committed:
Add 0.3.0 OLM bundle

operator-sdk olm-catalog gen-csv --csv-version=0.3.0 \
 --from-version=0.2.6 --update-crds --csv-channel=stable

cp deploy/olm-catalog/mobile-developer-console-operator/{0.2.6,0.3.0}/mdc_v1alpha1_mobileclient_crd.yaml

laurafitzgerald commented on deploy/olm-catalog/mobile-developer-console-operator/0.3.0/mobile-developer-console-operator.v0.3.0.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,270 @@
+    alm-examples: '[{"apiVersion":"mdc.aerogear.org/v1alpha1","kind":"MobileDeveloperConsole","metadata":{"name":"example-mdc"},"spec":{"oAuthClientId":"mobile-developer-console","oAuthClientSecret":"foobar"}}]'
+    capabilities: Basic Install
+  name: mobile-developer-console-operator.v0.3.0
+  namespace: placeholder
```
Is this okay? Should it be updated?

grdryn replied:
Yeah I think this is fine, the namespace(s) should be set through the OLM, right?

laurafitzgerald replied:
Yeah I was thinking that would be answer. Resolving.

laurafitzgerald commented on deploy/olm-catalog/mobile-developer-console-operator/0.3.0/mobile-developer-console-operator.v0.3.0.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,270 @@
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    alm-examples: '[{"apiVersion":"mdc.aerogear.org/v1alpha1","kind":"MobileDeveloperConsole","metadata":{"name":"example-mdc"},"spec":{"oAuthClientId":"mobile-developer-console","oAuthClientSecret":"foobar"}}]'
```
are we okay with the oAuthClientSecret being foobar here?

grdryn replied:
I don't think what's in here really matters, but I could be wrong. Is there something you'd suggest changing it to?

laurafitzgerald replied:
Yeah, I'm not really sure. I think it's fine for now anyway.

jjaferson replied:
I don't think it matters much and from the point of view of the integreatly operator it gets replaced by the openshift oauth client
